src/data/sum/order.lean

```diff
@@ -536,13 +534,7 @@ namespace with_bot
 `a`. -/
 def order_iso_punit_sum_lex : with_bot α ≃o punit ⊕ₗ α :=
 ⟨(equiv.option_equiv_sum_punit α).trans $ (equiv.sum_comm _ _).trans to_lex,
-begin
-  rintros (a | _) (b | _),
-  { simp },
-  { simp },
-  { simpa using with_bot.not_coe_le_bot a },
-  { simp }
-end⟩
+  by rintro (a | _) (b | _); simp; exact not_coe_le_bot _⟩
```

Comment:
Note: this creates a non-terminal `simp`.

Reply:
You can replace it with
```suggestion
  by rintro (a | _) (b | _); simp [some_eq_coe, none_eq_bot, not_coe_le_bot]⟩
```
if you care (because I believe this falls in the acceptable exceptions to non-terminal simps).

Reply:
I'll take your word on the acceptable exceptions.